### PR TITLE
Simplify and maintain build.py

### PR DESCRIPTION
**Pull request by sschulz92**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -28,29 +32,30 @@ def install():
...
-    setup.close()
-    tmpl.close()
+    with open("setup.tmpl", "r", encoding="utf-8") as tmpl:
+        tmpl_content = tmpl.read()
+        tmpl.close()
+    with open("setup.py", "w+", encoding="utf-8") as setup:
+        v = get_version()
+        setup.write(tmpl_content.format(
+            v, "{\n\t\t':python_version == \"2.7\"': ['futures']\n\t}"))
+        setup.close()
```
Comment (chadlwilson) [inline build.py:49]:
```suggestion
```

File: build.py [hunk 2]
```diff
@@ -28,29 +32,30 @@ def install():
...
 def generate_package():
     shutil.rmtree('dist', True)
     print('Creating getgauge package.')
     create_setup_file()
-    fnull = open(os.devnull, 'w')
-    call([sys.executable, 'setup.py', 'sdist'], stdout=fnull, stderr=fnull)
-    fnull.close()
+    with open(os.devnull, 'w', encoding="utf-8") as fnull:
+        call([sys.executable, 'setup.py', 'sdist'], stdout=fnull, stderr=fnull)
+        fnull.close()
```
Comment (chadlwilson) [inline build.py:58]:
```suggestion
```

**Review by chadlwilson — APPROVED** [review]

File: build.py [hunk 3]
```diff
@@ -28,29 +32,30 @@ def install():
...
-        v, "{\n\t\t':python_version == \"2.7\"': ['futures']\n\t}"))
-    setup.close()
-    tmpl.close()
+    with open("setup.tmpl", "r", encoding="utf-8") as tmpl:
+        tmpl_content = tmpl.read()
+        tmpl.close()
+    with open("setup.py", "w+", encoding="utf-8") as setup:
+        v = get_version()
+        setup.write(tmpl_content.format(
+            v, "{\n\t\t':python_version == \"2.7\"': ['futures']\n\t}"))
```
Comment (chadlwilson) [inline build.py:48]: Not really relevant for your PR, but this `python_version` declaration looks suspicious :-) Was added in 687657087ddc2df943afc88704e295e86d01ea3b so probably can go now.

Comment (sschulz92) [reply build.py]: I've removed the Python 2 related part. Is this what you had in mind too?

Comment (chadlwilson) [reply build.py]: Ahh yeah, thanks - was more of a note for later, but thanks for doing it. Probably need to remove this line too, as this is nothing in the formatting to supply it now.

https://github.com/getgauge/gauge-python/blob/97093f64da8890548f62b17bfeec4efd36b73bae/setup.tmpl#L21

**Pull request merged**